frontend/pages/admin/pacta-version/[id].vue

```diff
@@ -8,9 +8,27 @@ const { fromParams } = useURLParams()
 
 const id = presentOrCheckURL(fromParams('id'))
 
-const prefix = 'admin/pacta-version/[id]'
-const persistedPactaVersion = useState<PactaVersion>(`${prefix}.persistedPactaVersion`)
+const prefix = `admin/pacta-version/${id}`
 const pactaVersion = useState<PactaVersion>(`${prefix}.pactaVersion`)
+const { data: persistedPactaVersion, error, refresh } = await useAsyncData(`${prefix}.getPactaVersion`, () => {
```

Comment:
I like this (generally) but notice that almost all of this is standard/boilerplate.

proposal: can we come up with a wrapper for useAsyncData that does a few things:

1 - reuses the key between the loading OpKey (which we can do now that it isn't used for error reporting) and the useAsyncData key
2 - auto-wraps stuff in `withLoading` so that we don't need to have that ~anywhere
3 - adds the `handleOAPIError` if the return type of the promise can include the `api.Error` type
4 - automatically throws `createError` if an error is returned.
5 - ONLY hands back { data, refresh }, since error handling will all be handled standard.

If we have use cases where we need more customization folks can still choose to use `useAsyncData` in a way that is custom, but the utility described would cover 90% of our use cases with way less boilerplate.

Reply:
Good call, filed #22 to track. Oddly enough, the hardest part about this is the typings. `useAsyncData` has [hideously complicated typings](https://nuxt.com/docs/api/composables/use-async-data#type), and while I can just `Parameters` and `ReturnType` my way to matching them, the whole thing falls apart when you try to actually invoke `useAsyncData` and do stuff with the results.

It'll require some finessing to get it working.

Reply:
SGTM. I can take a crack at it.